inspirehep/modules/records/receivers.py

```diff
@@ -388,3 +388,14 @@ def generate_name_variations(sender, json, *args, **kwargs):
                 'output': full_name,
                 'payload': {'bai': bais[0] if bais else None}
             }})
+
+
+def populate_author_count(sender, json, *args, **kwargs):
+    """Populate author-count using only authors with 'inspi
+    re_roles' not equal to 'supervisor'."""
+    if 'hep.json' not in json.get('$schema'):
+        return
+
+    authors = json.get('authors', [])
+
+    json['author_count'] = len([author for author in authors if 'supervisor' not in author.get('inspire_roles', [])])
```

Comment:
Let's split this expression over multiple lines with explaining variables (http://wiki.c2.com/?IntroduceExplainingVariable).

Reply:
I agree! Does this look better 
```
authors_excluding_supervisors = [
    author for author in authors
    if 'supervisor' not in author.get('inspire_roles', [])
]
json['author_count'] = len(authors_excluding_supervisors)
```
?